### ISSUE TITLE
`NVFuserTest.CombinedReductionMultiPerBlock_CUDA` is failing

### ISSUE THREAD

**Issue by zasdfgbnm** [NVIDIA/Fuser#246]
```C++
[ RUN      ] NVFuserTest.CombinedReductionMultiPerBlock_CUDA
unknown file: Failure
C++ exception with description "Splitting an axis of non-Serial parallel type is not supported at this time. Parallelization strategy must be set after calling split.. Tensor: T8_g[ iblockIdx.y52{256}, ithreadIdx.x103{( ceilDiv(( ceilDiv(i2, 4) ), 8) )}, iS102{8}, ithreadIdx.y101{4} ]
Exception raised from split at /home/gaoxiang/Fuser/csrc/tensor_view.cpp:769 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::string) + 0x57 (0x7f45b91d5f57 in /home/gaoxiang/.local/lib/python3.10/site-packages/torch/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, std::string const&) + 0x64 (0x7f45b919aabb in /home/gaoxiang/.local/lib/python3.10/site-packages/torch/lib/libc10.so)
frame #2: nvfuser::TensorView::split(int, nvfuser::Val*, bool, bool) + 0x3fa (0x7f45b8e96bfa in /home/gaoxiang/Fuser/build/libnvfuser_codegen.so)
frame #3: nvfuser::TensorView::split(int, unsigned int, bool, bool) + 0xc1 (0x7f45b8e9b1c1 in /home/gaoxiang/Fuser/build/libnvfuser_codegen.so)
frame #4: <unknown function> + 0x5658f7 (0x55cdf692c8f7 in ./build/nvfuser_tests)
frame #5: <unknown function> + 0x5af71d (0x55cdf697671d in ./build/nvfuser_tests)
frame #6: <unknown function> + 0x5a092c (0x55cdf696792c in ./build/nvfuser_tests)
frame #7: <unknown function> + 0x5a1042 (0x55cdf6968042 in ./build/nvfuser_tests)
frame #8: <unknown function> + 0x5a1bda (0x55cdf6968bda in ./build/nvfuser_tests)
frame #9: <unknown function> + 0x5a358d (0x55cdf696a58d in ./build/nvfuser_tests)
frame #10: <unknown function> + 0x5a150e (0x55cdf696850e in ./build/nvfuser_tests)
frame #11: <unknown function> + 0x165f3a (0x55cdf652cf3a in ./build/nvfuser_tests)
frame #12: <unknown function> + 0x23790 (0x7f45b303c790 in /usr/lib/libc.so.6)
frame #13: __libc_start_main + 0x8a (0x7f45b303c84a in /usr/lib/libc.so.6)
frame #14: <unknown function> + 0x1a7a75 (0x55cdf656ea75 in ./build/nvfuser_tests)
" thrown in the test body.
[  FAILED  ] NVFuserTest.CombinedReductionMultiPerBlock_CUDA (7 ms)
```

**Comment by zasdfgbnm** [comment]
cc @liqiangxl This is failing on my machine starting from its very first commit https://github.com/NVIDIA/Fuser/commit/02501320abdc992177f5053ed989946d8d4d07a8. Is it failing on yours? If not, we might have some nondeterminism issue.

**Comment by zasdfgbnm** [comment]
Failing in this split:
https://github.com/NVIDIA/Fuser/blob/02501320abdc992177f5053ed989946d8d4d07a8/test/test_combined_inner_outer_reduction.cpp#L932

**Comment by naoyam** [comment]
The doesn't fail with A100, so it's likely to depend on the hardware parameters. Are you running it on 3090?

**Comment by zasdfgbnm** [comment]
I am running on 4090. On 3090, it works fine.

**Comment by naoyam** [comment]
It's getting warmer. Are you still running 3090 and 4090?

**Comment by zasdfgbnm** [comment]
> It's getting warmer. Are you still running 3090 and 4090?

Haha, yes, I have both.

**Comment by naoyam** [comment]
I don't see any machine with 4090 on dlcluster. It may be difficult to debug. I'm guessing the number of SMs and the register file size are affecting the heuristics in an unexpected way, so maybe we could reproduce the error on a different GPU by using the spec of 4090.

**Comment by zasdfgbnm** [comment]
I think you can reproduce it with manually setting `const int64_t device_multiprocessor_count = 128;`? You might not be able to run it, but it should reproduce the error.

**Comment by liqiangxl** [comment]
I'll take a look now.

**Comment by liqiangxl** [comment]
When `const int64_t device_multiprocessor_count =128;` the corresponding tidy = 4. So the outer reduction tv will split out a factor of {4} and that domain is parallelized by BIDy. When propagate parallelization from outer reduction,  `ithreadIdx.y65{4} in T11_l` is propagated to `T8_g[ iblockIdx.y52{256}, ithreadIdx.x103{( ceilDiv(( ceilDiv(i2, 4) ), 8) )}, iS102{8}, **ithreadIdx.y101{4}** ]`. The last domain in `T8_g` is from the vectorization of inner reduction tv and `T8_g` is calculated with the iner reduction part.
I modified to interface of `propagateParallelization` to avoid propagate from outer reduction to inner reduction. @naoyam actually asked this question during the code review but at that time I didn't think a case like this should happen as the inner reduciton and outer reduction are parallelized differently. However, this case shows it may happen.

**Comment by naoyam** [comment]
When is the problematic split called? Is it after (accidentally) propagating the parallelization of the outer tensors to `t8`?

**Comment by liqiangxl** [comment]
> When is the problematic split called? Is it after (accidentally) propagating the parallelization of the outer tensors to `t8`?
The error split is after the propagation, see last domain of T8_g, it is already parallelized by TIDy, which shouldn't happend.
`C++ exception with description "Splitting an axis of non-Serial parallel type is not supported at this time. Parallelization strategy must be set after calling split.. Tensor: T8_g[ iblockIdx.y52{256}, ithreadIdx.x103{( ceilDiv(( ceilDiv(i2, 4) ), 8) )}, iS102{8}, ithreadIdx.y101{4} ]`

**Comment by naoyam** [comment]
Is this where the split is called? 

https://github.com/NVIDIA/Fuser/blob/main/csrc/scheduler/normalization.cpp#L1823

I wonder why it splits the last ID. I don't quite remember how `T8` is scheduled, but isn't `ithreadIdx.y101{4}` split for vectorization?

**Comment by liqiangxl** [comment]
Yes. we do that split because we want to vectorize the load of partial reduction results.
(1) `i101{4}` in `T8` is propagated from `inner_reduction_tv->split(-1, vecx);`, it means we have 4 results of partial outer reducitons.
(2) then, for our reduction we have, `outer_reduction_tv->split(-1, tidy);     outer_reduction_tv->axis(-1)->parallelize(ParallelType::TIDy);` here in this case, `tidy = 4`, then the parallelization propagated to  `T8` and result in `ithreadIdx.y101{4}`, which shouldn't happen.

**Comment by naoyam** [comment]
Closed by #263
